### ISSUE TITLE
Problem with libgcc when statically linking simple program

### ISSUE THREAD

**Issue by JLague** [xalt/xalt#49]
I have an issue with `libgcc` when trying to statically link the following simple hello world program :
```
#include <stdio.h>

int
main() {
        printf("Hello world!\n");
}
```

I am building using the following command :
`gcc hello.c -o hello -static`

I get these errors :
```
11:46:33 XALT Tracing Activated for version: XALT 2.10.10

11:46:33 Starting ld

11:46:33 find_real_command: Searching for the real: ld.gold

11:46:33 find_real_command: found /cvmfs/soft.computecanada.ca/gentoo/2020/usr/bin/ld.gold

11:46:33 COMP_T: {"compiler":"gcc","compilerPath":"/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/bin/gcc","link_line":["gcc","test.c","-static"]}

11:46:33 SYSHOST: computecanada

11:46:33 WRKDIR: /tmp/billy_0924dae9-430b-4a41-830f-ab6005b92ce6_0udf9O

11:46:33 EPOCH 1622130393.5221

11:46:33 Using ld to find functions:  /cvmfs/soft.computecanada.ca/gentoo/2020/usr/bin/ld.gold -plugin /cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/libexec/gcc/x86_64-pc-linux-gnu/9.3.0/liblto_plugin.so -plugin-opt=/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/libexec/gcc/x86_64-pc-linux-gnu/9.3.0/lto-wrapper -plugin-opt=-fresolution=/tmp/cc8qeFAo.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_eh -plugin-opt=-pass-through=-lc -m elf_x86_64 -static /cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64/crt1.o /cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64/crti.o /cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/crtbeginT.o -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Compiler/intel2020/openmpi/4.0.3/lib/../lib64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../lib64 -L/cvmfs/soft.computecanada.ca/gentoo/2020/lib/../lib64 -L/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Compiler/intel2020/openmpi/4.0.3/lib -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/libfabric/1.10.1/lib -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/ucx/1.8.0/lib -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/imkl/2020.1.217/mkl/lib/intel64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/imkl/2020.1.217/lib/intel64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../.. -L/cvmfs/soft.computecanada.ca/gentoo/2020/lib -L/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib /tmp/ccb7cxpn.o --start-group -lgcc -lgcc_eh --end-group /cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/crtend.o /cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64/crtn.o --unresolved-symbols=report-all -o /dev/null

11:46:33 /cvmfs/soft.computecanada.ca/gentoo/2020/usr/bin/ld.gold -plugin /cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/libexec/gcc/x86_64-pc-linux-gnu/9.3.0/liblto_plugin.so -plugin-opt=/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/libexec/gcc/x86_64-pc-linux-gnu/9.3.0/lto-wrapper -plugin-opt=-fresolution=/tmp/cc8qeFAo.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_eh -plugin-opt=-pass-through=-lc -m elf_x86_64 -static /cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64/crt1.o /cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64/crti.o /cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/crtbeginT.o -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Compiler/intel2020/openmpi/4.0.3/lib/../lib64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../lib64 -L/cvmfs/soft.computecanada.ca/gentoo/2020/lib/../lib64 -L/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Compiler/intel2020/openmpi/4.0.3/lib -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/libfabric/1.10.1/lib -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/ucx/1.8.0/lib -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/imkl/2020.1.217/mkl/lib/intel64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/imkl/2020.1.217/lib/intel64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../.. -L/cvmfs/soft.computecanada.ca/gentoo/2020/lib -L/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib /tmp/ccb7cxpn.o --start-group -lgcc -lgcc_eh -lc --end-group /cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/crtend.o /cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64/crtn.o /tmp/billy_0924dae9-430b-4a41-830f-ab6005b92ce6_0udf9O/xalt.o /home/billy/xalt/xalt/lib64/xalt_initialize_regular.o /home/billy/xalt/xalt/lib64/librun_submission.a -ldl -lz -lm -lc   -t > /tmp/billy_0924dae9-430b-4a41-830f-ab6005b92ce6_0udf9O/link.txt  2>&1

11:46:33 Bad status when capturing tracing link output

11:46:33 /cvmfs/soft.computecanada.ca/gentoo/2020/usr/bin/ld.gold -plugin /cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/libexec/gcc/x86_64-pc-linux-gnu/9.3.0/liblto_plugin.so -plugin-opt=/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/libexec/gcc/x86_64-pc-linux-gnu/9.3.0/lto-wrapper -plugin-opt=-fresolution=/tmp/cc8qeFAo.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_eh -plugin-opt=-pass-through=-lc -m elf_x86_64 -static /cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64/crt1.o /cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64/crti.o /cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/crtbeginT.o -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Compiler/intel2020/openmpi/4.0.3/lib/../lib64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../lib64 -L/cvmfs/soft.computecanada.ca/gentoo/2020/lib/../lib64 -L/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Compiler/intel2020/openmpi/4.0.3/lib -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/libfabric/1.10.1/lib -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/ucx/1.8.0/lib -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/imkl/2020.1.217/mkl/lib/intel64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/imkl/2020.1.217/lib/intel64 -L/cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../.. -L/cvmfs/soft.computecanada.ca/gentoo/2020/lib -L/cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib /tmp/ccb7cxpn.o --start-group -lgcc -lgcc_eh -lc --end-group /cvmfs/soft.computecanada.ca/easybuild/software/2020/Core/gcccore/9.3.0/lib/gcc/x86_64-pc-linux-gnu/9.3.0/crtend.o /cvmfs/soft.computecanada.ca/gentoo/2020/usr/lib/../lib64/crtn.o /tmp/billy_0924dae9-430b-4a41-830f-ab6005b92ce6_0udf9O/xalt.o /home/billy/xalt/xalt/lib64/xalt_initialize_regular.o /home/billy/xalt/xalt/lib64/librun_submission.a -ldl -lz -lm -lc
util/compute_sha1.c:43: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
util/compute_sha1.c:49: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
util/build_uuid.c:25: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
util/build_uuid.c:30: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
../stdlib/strtod_l.c:309: error: undefined reference to '__multf3'
../stdlib/strtod_l.c:353: error: undefined reference to '__addtf3'
../stdlib/strtod_l.c:193: error: undefined reference to '__multf3'
../stdlib/strtod_l.c:205: error: undefined reference to '__multf3'
../stdlib/strtod_l.c:205: error: undefined reference to '__multf3'
collect2: error: ld returned 1 exit status
```

I have tried building another, more complex, statically linked executable (BEDOPS) and everything worked without any problems.

Appending `-lgcc` to the `$SYSLIBS` variable in the `ld` wrapper fixes the issue, but I don't know if it's the best way to do it since it looks like the issue is coming from somewhere else.

**Comment by rtmclay** [comment]
I have modified the ld wrapper script to include "-lgcc" when the user has -static on the command line.  Please test xalt-2.10.12 to see if this handles this issue.

**Comment by JLague** [comment]
My issue is fixed with 2.10.12. Thanks!
